test/Templates.Test/Helpers/AddFirewallExclusion.cs

```diff
@@ -0,0 +1,70 @@
+﻿using System;
+using System.Diagnostics;
+using System.IO;
+
+namespace Templates.Test.Helpers
+{
+    internal class AddFirewallExclusion : IDisposable
+    {
+        private bool _disposedValue = false;
+        private readonly string _exclusionPath;
+
+        public static IDisposable Create(string exclusionPath)
+        {
+            return new AddFirewallExclusion(exclusionPath);
+        }
+
+        private AddFirewallExclusion(string exclusionPath)
+        {
+            if (!File.Exists(exclusionPath))
+            {
+                throw new FileNotFoundException();
+            }
+
+            _exclusionPath = exclusionPath;
+            var startInfo = new ProcessStartInfo
+            {
+                RedirectStandardError = true,
+                RedirectStandardOutput = true,
+                CreateNoWindow = true,
+                FileName = "cmd.exe",
+                Arguments = $"/c netsh advfirewall firewall add rule name=\"Allow {exclusionPath}\" dir=in action=allow program=\"{exclusionPath}\"",
```

Comment:
Out of interest, does this require you to be running as Administrator? Do our tests already fail if you're not Administrator?

Reply:
Or to be clearer, I suppose the existence of this code implies our tests aren't already running as Administrator, because if they were then this would not be needed.

My only concern would be if this could sometimes pop up the Windows UAC dialog but I suppose it must not do because otherwise the tests would always fail.

Reply:
OK ignore these comments. I see from your initial PR description how this works. Looks good.

Reply:
Yes it requires you to be running the process as an admin (either in VS or cmd line) and we don't do that in our CI builds. But that's okay since the existence of this code does not lead to failures if not running as admin. This is just useful when building locally.

Question for you @SteveSandersonMS how come building release/2.0 branch does not have the firewall prompts? What changed since then that is causing the firewall warnings in the first place?

Reply:
I'm not aware of anything we've changed in our source code that would affect this. Does it definitely *not* produce the prompts now if we build `release/2.0` still now? I thought maybe the CI machines have changed in some way that has reset whatever TCP-listening permission it needs.

Reply:
I meant while building locally. It does not show the prompts for release/2.0 at all.